CMakeLists.txt

```diff
@@ -81,11 +81,54 @@ endif()
 # Do we want to build in Python support where available
 #
 option(fletch_BUILD_WITH_PYTHON "Build with Python support where appropriate" FALSE)
+
+set(fletch_PYTHON_MAJOR_VERSION "2" CACHE STRING
+    "Python version to use: 3, 2, or empty")
+set_property(CACHE fletch_PYTHON_MAJOR_VERSION PROPERTY STRINGS "3" "2")
+
 if (fletch_BUILD_WITH_PYTHON)
 
-  find_package(PythonInterp 2.7 REQUIRED)
-  find_package(PythonLibs 2.7 REQUIRED)
+  # If we change python versions re-find the bin, include, and libs
+  if (NOT _prev_fletch_pymajor_version STREQUAL fletch_PYTHON_MAJOR_VERSION)
+    # but dont clobber initial settings in the instance they are specified via
+    # commandline (e.g  cmake -DPYTHON_EXECUTABLE=/my/special/python)
+    if (_prev_fletch_pymajor_version)
+      message(STATUS "The Python version changed; refinding the interpreter")
+      message(STATUS "Previous python version was ${_prev_fletch_pymajor_version}")
+      unset(_prev_fletch_pymajor_version CACHE)
+      unset(PYTHON_EXECUTABLE CACHE)
+      unset(PYTHON_INCLUDE_DIR CACHE)
+      unset(PYTHON_LIBRARY CACHE)
+      unset(PYTHON_LIBRARY_DEBUG CACHE)
+    endif()
+  endif()
+
+  if (fletch_PYTHON_MAJOR_VERSION STREQUAL "3")
+    find_package(PythonInterp 3.4 REQUIRED)
+    find_package(PythonLibs 3.4 REQUIRED)
+  else()
+    find_package(PythonInterp 2.7 REQUIRED)
+    find_package(PythonLibs 2.7 REQUIRED)
+  endif()
+
+  # Check to ensure that the python executable agrees with the major version
+  execute_process(
+    COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; print(sys.version[0:3])"
+    RESULT_VARIABLE _exitcode
+    OUTPUT_VARIABLE _python_version)
+  if(NOT ${_exitcode} EQUAL 0)
+    message(FATAL_ERROR "Python command to get version failed with error code: ${_exitcode}")
+  endif()
+  # Remove supurflous newlines (artifacts of print)
+  string(STRIP "${_python_version}" _python_version)
+
+  if(NOT _python_version MATCHES "^${fletch_PYTHON_MAJOR_VERSION}.*")
+    message(FATAL_ERROR "Requested python \"${fletch_PYTHON_MAJOR_VERSION}\" but got \"${_python_version}\"")
+  endif()
 
+  # Make a copy so we can determine if the user changes python versions
+  set(_prev_fletch_pymajor_version "${fletch_PYTHON_MAJOR_VERSION}" CACHE INTERNAL
+    "allows us to determine if the user changes python version")
```

Comment:
Should we write out the python version used to the fletchConfig.cmake? It seems like kwiver or other consumers might need to know which version of python was used to build the TPL stuff.

Reply:
Definitely. I wasn't aware of what fletch_CONFIG_INPUT was before, but now I see. In fact, we should likely do that with all the other SELECT_VERSION variables.

Reply:
I added the suggested change in this branch. As for the other change I mentioned, I made a separate PR #231 for PR modularity.